File: csharp_body_deref/Scripts/CustomLogic/CustomLogicFieldExpressionAst.cs
// ==================== AoTTG2 cross-reference ====================
// Type: CustomLogic.CustomLogicFieldExpressionAst
// Update status: unchanged since the previous dump
// Real method bodies (native C decompile): source/cpp/CustomLogic/CustomLogicFieldExpressionAst.c
// Prior real C# source (older reference): Assets/Scripts/CustomLogic/Ast/CustomLogicFieldExpressionAst.cs
// Bodies below are empty IL2CPP metadata stubs; the logic lives in the native C.
// ===============================================================
using Il2CppDummyDll;

namespace CustomLogic;

[Token(Token = "0x2000291")]
internal class CustomLogicFieldExpressionAst : CustomLogicBaseExpressionAst
{
	[Token(Token = "0x4000DD2")]
	[FieldOffset(Offset = "0x18")]
	public string FieldName;

	[Token(Token = "0x4000DD3")]
	[FieldOffset(Offset = "0x20")]
	public CustomLogicBaseExpressionAst Left;

	[Token(Token = "0x6000F7C")]
	[Address(RVA = "0x3FB8A00", Offset = "0x3FB8A00", VA = "0x3FB8A00")]
	public CustomLogicFieldExpressionAst(string name, int line)
	{
	}
}

File: cpp/CustomLogic/CustomLogicFieldExpressionAst.c
// Type: CustomLogic.CustomLogicFieldExpressionAst
// Ghidra (System V/gcc) decompilation, IL2CPP structs + signatures applied.
// ---- AoTTG2 cross-reference ----
// Update status: unchanged since the previous dump
// C# structure: source/csharp/Scripts/CustomLogic/CustomLogicFieldExpressionAst.cs
// Prior real C# source (older reference): Assets/Scripts/CustomLogic/Ast/CustomLogicFieldExpressionAst.cs
// --------------------------------

// CustomLogic.CustomLogicFieldExpressionAst$$.ctor
// il2cpp: void CustomLogic_CustomLogicFieldExpressionAst___ctor (CustomLogic_CustomLogicFieldExpressionAst_o* __this, System_String_o* name, int32_t line, const MethodInfo* method);
// 0x40b8a00

void CustomLogic_CustomLogicFieldExpressionAst___ctor
               (CustomLogic_CustomLogicFieldExpressionAst_o *__this,System_String_o *name,int32_t line,
               MethodInfo *method)

{
  System_Object___ctor((Il2CppObject *)__this,(MethodInfo *)0x0);
  (__this->fields).Type = 6;
  (__this->fields).Line = line;
  (__this->fields).FieldName = name;
  il2cpp_runtime_helper_022b4080(&(__this->fields).FieldName,name);
  return;
}


